From 8ba863f06c89a995c8bfb6d56a8ef6e668ae2808 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:52 -0400
Subject: [usb] wusbcore: set pointers to NULL after freeing in error cases

Message-id: <1410887463-20674-101-git-send-email-dzickus@redhat.com>
Patchwork-id: 95488
O-Subject: [RHEL7 PATCH 100/591] usb: wusbcore: set pointers to NULL after freeing in error cases
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 6741448eb01a1844b78c83fa8faf0c95524fc09b
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Thu Sep 26 14:08:16 2013 -0500

    usb: wusbcore: set pointers to NULL after freeing in error cases

    This patch fixes two cases where error handling code was freeing memory
    but not setting the pointer to NULL.  This could lead to a double free
    in the HWA shutdown code.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-xfer.c b/drivers/usb/wusbcore/wa-xfer.c
index 61b0597..0b27146 100644
--- a/drivers/usb/wusbcore/wa-xfer.c
+++ b/drivers/usb/wusbcore/wa-xfer.c
@@ -1560,6 +1560,7 @@ error_submit_buf_in:
    xfer, seg_idx, result);
  seg->result = result;
  kfree(wa->buf_in_urb->sg);
+ wa->buf_in_urb->sg = NULL;
 error_sg_alloc:
  __wa_xfer_abort(xfer);
 error_complete:
@@ -1859,6 +1860,7 @@ out:
 
 error_dti_urb_submit:
  usb_put_urb(wa->buf_in_urb);
+ wa->buf_in_urb = NULL;
 error_buf_in_urb_alloc:
  usb_put_urb(wa->dti_urb);
  wa->dti_urb = NULL;
-- 
1.7.1